src/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,277 @@
+     * @var DownloaderInterface
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $downloader;
+
+    /**
+     * Only valid when examined from within doExecute().
+     *
+     * @var Filesystem
+     *
+     * @psalm-suppress PropertyNotSetInConstructor
+     */
+    private $filesystem;
+
+    /**
+     * @param string $pharFile The location of the execed phar file.
+     */
+    public function __construct(string $pharFile, ?DownloaderInterface $downloader = null)
+    {
+        parent::__construct();
+
+        $this->pharFile = $pharFile;
+
+        if ($downloader !== null) {
+            $this->downloader = $downloader;
+        }
+    }
+
+    protected function configure(): void
+    {
+        parent::configure();
+
+        $this->setName('self-update')->setDescription('Updates the phpcq phar file');
+
+        $this->addOption(
+            'cache',
+            'x',
+            InputOption::VALUE_REQUIRED,
+            'Path to the phpcq cache directory',
+            (getenv('HOME') ?: sys_get_temp_dir()) . '/.cache/phpcq'
+        );
+
+        $this->addOption(
+            'channel',
+            null,
+            InputOption::VALUE_REQUIRED,
+            'The channel of the release. Right now only unstable is available',
+            'unstable'
+        );
+
+        $this->addOption(
+            'base-uri',
+            null,
+            InputOption::VALUE_REQUIRED,
+            'The base uri of the phpcq releases',
+            'https://phpcq.github.io/distrib/phpcq'
+        );
+
+        $this->addOption(
+            'unsigned',
+            null,
+            InputOption::VALUE_NONE,
+            'Disable signature checking'
+        );
+
+        $this->addOption(
+            'dry-run',
+            null,
+            InputOption::VALUE_NONE,
+            'Do not perform update, only check for a new release'
+        );
+
+        $this->addOption(
+            'force',
+            'f',
+            InputOption::VALUE_NONE,
+            'Force to update'
+        );
+    }
+
+    protected function prepare(InputInterface $input): void
+    {
+        parent::prepare($input);
+
+        /** @psalm-suppress DocblockTypeContradiction */
+        if ($this->downloader === null) {
+            $cachePath        = $this->getCachePath();
+            $this->downloader = new FileDownloader($cachePath, $this->config->getAuth());
+        }
+
+        $this->filesystem = new Filesystem();
+    }
+
+    protected function doExecute(): int
+    {
+        $baseUri          = $this->getBaseUri();
+        $installedVersion = $this->getInstalledVersion();
+        $availableVersion = substr($this->downloader->downloadFile($baseUri . '/current.txt', '', true), 6);
+
+        if (! $this->shouldUpdate($installedVersion, $availableVersion)) {
+            return 0;
+        }
+
+        $pharUrl = $baseUri . '/phpcq.phar';
+        $downloadedPhar = tempnam(sys_get_temp_dir(), 'phpcq.phar-');
```

Comment:
Shall we rather use the cache dir here? Just an idea but no requirement.

Reply:
We already use the sys temp dir for none cache related stuffs, e.g. https://github.com/phpcq/phpcq/blob/2f08141ba21974fa22fc62aff2693e19b249fb1e/src/Command/AbstractCommand.php#L166.

I don't want to pollute the cache dir with that. The file is already in the cache, in fact getting the path of the file in the cache would be enough here.

Reply:
Fine with me.